src/Web/Scim/Schema/User.hs

```
@@ -242,6 +246,9 @@ instance FromJSON NoUserExtra where
 instance ToJSON NoUserExtra where
   toJSON _ = object []
 
+instance Patchable NoUserExtra where
+  applyOperation a _ = pure a
```

should this silently do nothing?  or rather reject all attempts at being patched with a `Left`?

no, wait: this is just a different, redundant haskell representation of the json empty object.  what's this good for?  why is it important enough to destroy the isomorphism property of serialization (now there are two haskell representations for the same `{}`)?  can you add that to the haddocks of the type?

---

the "what is it good for" is   "it was already there"

I think I agree this should be a `Left` not a `Right`.  There are no paths that lead to a valid `PATCH` for a type with no fields

---

>  why is it important enough to destroy the isomorphism property of serialization (now there are two haskell representations for the same {})

??  Having two types have the same serializaiton does not destroy any properties. There's an infinite amount of json types with the same serialization. I'm not sure what you mean by this sentence.

---

serialization is not an isomorphism any more then: serializing and then de-serializing may give you something different from the input.

for once, that makes roundtrip testing awkward.  i have a feeling it also makes reasoning about production code harder sometimes, but i can't think of an example right now.

anyway, not a blocker.

---

I still don't understand. What exactly is not Roundtripping about `FromJSON/ToJSON` of `UserExtra`?

---

I think the point is that in this case data serialized from one type can be de-serialized into a different one. Generally, the downsides of this could be (by accident) successfully parsing something of a different type as intended or as a human not being able to say what type some random JSON was serialized from.

I agree with Arian though, this is not a problem for the roundtrip property. It also happens all the time already, e.g. `Nothing :: Maybe Foo` and `Nothing :: Maybe Bar` will always be serialized identically.

---

I don't think this is resolved yet. I agree with @fisx 's observation that this function should return `throwError`  instead of silently accepting everything

---

This instance now throws error, hopefully the message is helpful enough.